File: CodeMag/AjaxAndRest/WebAPI/EntityClasses/Product.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebAPI
{
    [Table("product", Schema="production")]
    public partial class Product
    {
        [Required]
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int productid {get; set; }
        [Required(ErrorMessage = "The Product Name is required")]
        public string name { get; set; }
        [Required]
        public string productnumber { get; set; }
        public string color { get; set; } 
        [Required]
        public decimal standardcost { get; set; }
        [Required]
        public decimal listprice { get; set; }
        public string size { get; set; }
        public decimal? weight { get; set; }
        public int? productsubcategoryid { get; set; }
        public int? productmodelid { get; set; }
        [Required]
        public DateTime sellstartdate { get; set; }
        public DateTime? sellenddate { get; set; }
        public DateTime? discontinueddate { get; set; }
        [Required]
        public Guid rowguid { get; set; }
        [Required]
        public DateTime modifieddate { get; set; }
        [Required]
        public int safetystocklevel { get; set; }
        [Required]
        public int reorderpoint { get; set; }
        [Required]
        public int daystomanufacture { get; set; }
    }

}

File: CodeMag/AjaxAndRest/AjaxSample/wwwroot/js/product.js
function getFromInput() {
    return {
        "productid": GetValue("productid"),
        "name": GetValue("name"),
        "productnumber": GetValue("productnumber"),
        "color": getValue("color"),
        "safetystocklevel": getValue("safetystockvalue"),
        "reorderpoint": getValue("reorderpoint"),
        "standardcost": getValue("standardcost"),
        "listprice": getValue("listprice"),
        "size": getValue("size"),
        "weight": getValue("weight"),
        "daystomanufacture": getValue("daystomanufacture"),
        "sellstartdate": new Date(getValue("sellstartdate")),
    }
}

function setInput() {
    setValue("productid", product.productid);
    setValue("name", product.name);
    setValue("productnumber", product.productnumber);
    setValue("color", product.color);
    setValue("safetystocklevel", product.safetystocklevel);
    setValue("reorderpoint", product.reorderpoint);
    setValue("standardcost", product.standardcost);
    setValue("listprice", product.listprice);
    setValue("size", product.size);
    setValue("weight", product.weight);
    setValue("daystomanufacture", product.daystomanufacture);
    setValue("sellstartdate", product.sellstartdate);
}

function clearInput() {
    setValue("productid", "0");
    setValue("name", "");
    setValue("productnumber", "");
    setValue("color", "");
    setValue("safetystocklevel", "0");
    setValue("reorderpoint", "0");
    setValue("standardcost", "0");
    setValue("listprice", "0");
    setValue("size", "0");
    setValue("weight", "0");
    setValue("daystomanufacture", "0");
    setValue("sellstartdate", new Date().toDateString());

}